integration_tests/run_pyspark_from_build.sh

```diff
@@ -174,8 +174,8 @@ else
     else
         TEST_PARALLEL_OPTS=("-n" "$TEST_PARALLEL")
     fi
-    RUN_DIR=${RUN_DIR-"$SCRIPTPATH"/target/run_dir}
-    mkdir -p "$RUN_DIR"
+
+    RUN_DIR=${RUN_DIR-`mktemp -p "$SCRIPTPATH"/target -d run_dir-$(date +%Y%m%d%H%M%S)-XXXX`}
```

Comment:
Let us stay consistent with regard to command output substitution  backticks vs  `${cmd}`, and the latter is [preferred][1]

Realized that my prev suggestion is not backwards-compatible with the prior behavior for externally-supplied `RUN_DIR` env.  Let us add `-u` to mktemp and put
 and restore the line  `mkdir -p "$RUN_DIR"`. SInce mktemp -u will not have the side-effect of creating dir any more we can safely use it in an intermediate variable, so you might not even need nested command substitution in RUN_DIR assignment.

[1]: http://mywiki.wooledge.org/BashFAQ/082